lib/rspec/core/example_group.rb

```diff
@@ -253,11 +273,9 @@ def self.define_example_group_method(name, metadata={})
 
           begin
             description = args.shift
-            combined_metadata = metadata.dup
-            combined_metadata.merge!(args.pop) if args.last.is_a? Hash
-            args << combined_metadata
-
-            subclass(self, description, args, registration_collection, &example_group_block)
+            combined_metadata = metadata.merge(Metadata.build_hash_from(args))
```

Comment:
This is arguably less complicated, partially due to 1.8.7 support, it could have been `|description=nil, *args, &example_group_block|`.
However, `shift`/`unshift` might still be necessary, since there's nothing that prevents the `description` to be a `Symbol`, and `build_hash_from` would consider it as metadata keys.